src/puppetlabs/puppetdb/query/events.clj

```diff
@@ -50,49 +50,74 @@
 (defn with-latest-events
   "CTE to wrap unioned queries when distinct_resources is used"
   [query]
-  (format
-    "WITH latest_events AS
-           (SELECT certnames.certname,
-                   resource_events.resource_type,
-                   resource_events.resource_title,
-                   resource_events.property,
-                   MAX(resource_events.timestamp) AS timestamp
-                   FROM resource_events
-                   JOIN certnames ON resource_events.certname_id = certnames.id
-                   WHERE resource_events.timestamp >= ?
-                   AND resource_events.timestamp <= ?
-                   GROUP BY certname, resource_type, resource_title, property)
-           %s" query))
+  (let [collate (if (sutils/postgres?) "COLLATE \"C\"" "")]
+    (format
+      "WITH latest_events AS
+       (SELECT certname,
+       configuration_version,
+       start_time as run_start_time,
+       end_time as run_end_time,
+       receive_time as report_receive_time,
+       hash,
+       status,
+       distinct_events.timestamp AS timestamp,
+       distinct_events.resource_type AS resource_type,
+       distinct_events.resource_title AS resource_title,
+       distinct_events.property as property,
+       new_value,
+       old_value,
+       message,
+       file,
+       line,
+       containment_path,
+       containing_class,
+       name
+       FROM
+       (SELECT certname_id,
+       resource_type %s AS resource_type,
+       resource_title %s AS resource_title,
+       property %s AS property,
+       MAX(resource_events.timestamp) AS timestamp
```

Comment:
well, truly it should be distinct_events, but then we'd have to realias it as timestamp above to get the return field right, so we'd have
```
distinct_events.distinct_events_timestamp as timestamp
```
up on line 63. I could drop the aliasing there at that point but I don't see the benefit really. Does that seem desirable?